src/Domain/Entity/CoreEntity.php

```diff
@@ -18,6 +18,9 @@
     /** @var Pid */
     protected $pid;
 
+    /** @var string */
+    protected $type = '';
```

Comment:
What do you reckon to these being protected class constants? Then we wouldn't end up creating new strings for every entity we create (micro-memory optimisations)